File: matlab/lib/+Ice/ParseException.m
% ParseException   Summary of ParseException
%
% This exception is raised if there was an error while parsing a string.
%
% ParseException Properties:
%   str - Describes the failure and includes the string that could not be parsed.

% Copyright (c) ZeroC, Inc. All rights reserved.

classdef ParseException < Ice.LocalException
    properties
        % str - Describes the failure and includes the string that could not be parsed.
        str char
    end
    methods
        function obj = ParseException(errID, msg, str)
            if nargin <= 2
                str = '';
            end
            if nargin == 0 || isempty(errID)
                errID = 'Ice:ParseException';
            end
            if nargin < 2 || isempty(msg)
                msg = 'Ice.ParseException';
            end
            obj = obj@Ice.LocalException(errID, msg);
            obj.str = str;
        end
    end
end

File: java/src/Ice/src/main/java/com/zeroc/Ice/ParseException.java
// Copyright (c) ZeroC, Inc.

package com.zeroc.Ice;

/** Reports a failure that occurred while parsing a string. */
public final class ParseException extends LocalException {
    public ParseException(String message) {
        super(message);
    }

    public ParseException(String message, Throwable cause) {
        super(message, cause);
    }

    public String ice_id() {
        return "::Ice::ParseException";
    }

    private static final long serialVersionUID = 7926520612608626840L;
}
